### PR TITLE
Use flutter gcp project for linux tasks.

### PR DESCRIPTION
**Pull request by godofredoc**
Use flutter-cirrus for Cirrus Linux tasks in the plugins repository.

Bug: https://github.com/flutter/flutter/issues/77624

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by stuartmorgan** [comment]
The failing test is because the driver doesn't support Chrome 89 yet, and anything that causes the image to be regenerated will get 89 currently. (Last time I looked I didn't see an obvious way to get a pinned version in the Docker file).

I need to check on what the plan is for web integration tests on boots though; that specific test will be removed shortly.

**Comment by stuartmorgan** [comment]
I've updated the integration test step, and was able to successfully run it with the newer Docker image after your driver change, so if you rebase this it should hopefully be green.

File: .cirrus.yml [hunk 1]
```diff
@@ -110,16 +116,20 @@ task:
       build_script:
         - flutter config --enable-linux-desktop
         - ./script/incremental_build.sh build-examples --linux
-      test_script:
```
Comment (stuartmorgan) [inline .cirrus.yml:113]: Why is this removed? Separating these is helpful in looking at results.

Comment (godofredoc) [reply .cirrus.yml:113]: Nice catch, merge conflict editing problem. Updating!

**Review by stuartmorgan — APPROVED** [review]
LGTM otherwise.

**Comment by stuartmorgan** [comment]
Thanks for driving this! Looking forward to the improved throughput on presubmits :)
